READMEs/[[...slug]].md

```diff
@@ -0,0 +1,107 @@
+
+- The use of single brackets (e.g. `[slug].tsx`) enables the matching of the current segment. So this would match `localhost:3000/page-1` or `localhost:3000/page-2`. Importantly, it wouldn't match our page in question.
+- The use of an ellipsis (e.g. `[...slug].tsx`) enables the match to be a catch-all match. So, this would match `localhost:3000/page-1` OR `localhost:3000/path/to/page-1`. This _would_ match the path in question, and every other path _except one_.
+- The path we wouldn't match is `localhost:3000` (the homepage). In order to accommodate that, we use double brackets with ellipsis (e.g. `[[...slug]].tsx`), which enables the catch-all segment to be _optional_.
+
+### Considerations
+
+It's useful to point out that we could, conceivably, have multiple files rather than a single `[[...slug]].tsx` file. For example, a site like ours that builds from CMS content might have this structure in its `pages` directory:
+
+```
+|__pages
+  |
+  |__stories
+    |
+    |__[[...slug]].tsx
+  |
+  |__events
+    |
+    |__[[...slug]].tsx
+```
+
+But, importantly, this approach would mean two things:
+
+1. Next.js file structure would determine the information architecture, at least as it relates to page paths. In our approach, we define the paths in the CMS.
+2. This would require that news stories all have the wildcard portion at the end of the path (e.g. `/stories/some-story-title` or `/stories/path/to/some-story-title`). In our setup, this is not the case (e.g. we have `/[VAMC system]/stories/[story-title]`).
+
+### From URL to HTML
+
+As we've seen, we accommodate our use case by implementing a single entry point, of sorts. So, regardless of the URL of the page in question, the same top-level page file is used. So how do we build different types of pages?
+
+There are two main steps:
+
+1. The first step takes place inside `getStaticProps`. When we want to generate HTML for a given URL, we first ping the CMS for more information about the page at that URL. Since the CMS knows the URL for a given page, it can look up the entity based on the URL and provide us important information about the entity. Most importantly, it can tell us what _type_ of entity it represents.
+2. Once we know the type of the entity, we can then pass the data object to the appropriate template. We see this at the top of the `[[...slug]].tsx` file in the default export. Our top-level page templates inside the `pages` directory need to export a template/component that can generate the HTML for the page. Here, we switch on the entity type, and return the appropriate component.
+
+And that is basically it! `[[...slug]].tsx` isn't magic, but it is the entry point to the entire process, and it's a critical piece to understanding the architecture of this project.
```

Comment:
i would bold `the entry point to the entire process, and it's a critical piece to understanding the architecture of this project.` because yes.